### PR TITLE
Delete item button

### PR DESCRIPTION
**Pull request by CindySchalit**
STOP! Hammer time.

* Have you written unit tests?
* Reviewer: have you actually reviewed the code?

Able to delete item from cart upon pushing delete button.  Deletes one item at a time, i.e. if there are 4 inflatable turtles in the cart, removes only one turtle - other 3 turtles stay in cart.  Cart page refreshes to reflect updated cart.

Would love to team up with someone who is more comfortable with testing to write the tests ... AP, AP, AP?  =)

### CODE REVIEW

File: app/components/ManageCart/reducers/reducer.js [hunk 1]
```diff
@@ -74,10 +73,10 @@ export const addToCart = (item) => (dispatch) => {
   .catch(console.error.bind(console))
 }
 
-export const removeFromCart = (item, cartId) => (dispatch) => {
+export const removeFromCart = (item) => (dispatch) => {
   axios.put(`/api/cart/products/sub`, item)
   .then( res => {
-    if (res.data === 'OK') dispatch(removeItem(item))
+    if (res.data === 'OK') dispatch(retrieveItems())
```
Comment (DillGromble) [inline app/components/ManageCart/reducers/reducer.js:79]: Dispatching retrieveItems is a clever workaround for effectively using the backend logic for removing an item!  
I was actually pondering this problem when I looked at this, and I had never thought of that solution. 

However, it does render completely moot the REMOVE_ITEM action creator, reducer case, and constant.

File: app/components/ManageCart/reducers/reducer.js [hunk 2]
```diff
@@ -43,11 +43,10 @@ const reducer = (state = initialState, action) => {
     return newState
 
   case REMOVE_ITEM:
-    const index = newState.cartItems.findIndex(action.item)
-    newState.cartItems.slice(index, 1)
+    newState.currentCart.products = state.currentCart.products.filter(prod => prod !== action.item)
```
Comment (glebec) [inline app/components/ManageCart/reducers/reducer.js:46]: If the items have IDs, it is better to filter by ID than object equality. Imagine you get a response from the backend, the object created won't be the same object you already have in memory.

**Review by glebec — COMMENTED** [review]
If you keep the re-retrieval of all items as your solution, this becomes moot. However, I would say that selectively removing a single item is more direct and performant.

File: app/components/ManageCart/react/ManageCart.js [hunk 3]
```diff
@@ -1,39 +1,48 @@
...
-          </div>
-        ))}
-      </ul>
+  deleteClickedItem(item) {
+    this.props.removeItem(item)
+  }
+
+  render() {
+    const userName = this.props.userName
+    const itemsInCart = this.props.itemsInCart
```
Comment (glebec) [inline app/components/ManageCart/react/ManageCart.js:16]: Good place to use destructuring.

File: app/components/ManageCart/react/ManageCart.js [hunk 4]
```diff
@@ -1,39 +1,48 @@
...
+          <div className="col-xs-4"><h3>Quantity</h3></div>
+        </div>
+          {itemsInCart && itemsInCart.map(item => (
+            <div key={item.id}>
+                <div className="col-xs-4">{item.name}</div>
+                <div className="col-xs-4">${item.price}</div>
+                <div className="col-xs-4">{item.quantity}</div>
+                <button
+                  className="btn btn-default btn-xs"
+                  onClick={this.deleteClickedItem.bind(this, item)}
```
Comment (glebec) [inline app/components/ManageCart/react/ManageCart.js:33]: are you using `bind` for partial function application? You've already bound this function in the constructor; I'd say an arrow is perhaps less verbose in this case, e.g. `() => this.deleteClickedItem(item)`.

**Comment by DillGromble** [comment]
This PR was merged into #81

**Pull request closed without merging**